tests/foreman/cli/test_computeresource_vmware.py

```diff
@@ -146,6 +147,34 @@ def test_positive_provision_end_to_end(
         != 'Pending installation',
         timeout=1800,
         delay=30,
+        silent_failure=True,
+        handle_exception=True,
     )
     host_info = sat.cli.Host.info({'id': host['id']})
     assert host_info['status']['build-status'] == 'Installed'
+
+    # Perform OS version check and check if root password is properly updated
+    provisioning_host = ContentHost(
+        host_info['network-interfaces'][0]['ipv4-address'], auth=module_ssh_key_file
+    )
+    expected_rhel_version = host_info['operating-system']['operating-system'][-3:]
```

Comment:
You are setting yourself for a failure with this kind of version parsing
```
In [1]: '8.10'[-3:]
Out[1]: '.10'
```
also, that field might contain os name + version + some suffix, like `beta` or possibly `eus`.
I'd suggest to use nailgun to get the host info, which should resolve the os entity where you can directly access `major` and `minor` versions.